precise/iaf_psc_exp_ps_lossless.h

```diff
@@ -0,0 +1,486 @@
+    const long lag,
+    const double t0,
+    const double dt );
+
+  /**
+   * Emit a single spike at a precisely given time.
+   *
+   * @param origin        Time stamp at beginning of slice
+   * @param lag           Time step within slice
+   * @param spike_offset  Time offset for spike
+   */
+  void emit_instant_spike_( const Time& origin,
+    const long lag,
+    const double spike_offset );
+
+  /**
+   * Localize threshold crossing by bisectioning.
+   * @param   double length of interval since previous event
+   * @returns time from previous event to threshold crossing
+   */
+  double bisectioning_( const double dt ) const;
+
+  /**
+   * Retrospective spike detection by state space analysis.
+   * The state space spanning the non-spiking region is bound by the following
+   * system of inequalities:
+   * threshold line V < \theta, envelope, V < b(I_e), line corresponding to the
+   * final timestep
+   * V < f(h, I) (or) linear approximation of the envelope, V < g(h, I_e).
+   * The state space spanning the spiking region is bound by the following
+   * system of inequalities:
+   * threshold line V < \theta, envelope, V > b(I_e) and line corresponding to
+   * the final timestep
+   * V > f(h, I) (or) linear approximation of the envelope, V < g(h, I_e).
+   * Propagate the neuron's state by dt.
+   * @returns time to emit spike.
+   */
+  double is_spike_( const double );
+
+  // ----------------------------------------------------------------
+
+  /**
+   * Independent parameters of the model.
+   */
+  struct Parameters_
+  {
+    /** Membrane time constant in ms. */
+    double tau_m_;
+
+    /** Time constant of exc. synaptic current in ms. */
+    double tau_ex_;
+
+    /** Time constant of inh. synaptic current in ms. */
+    double tau_in_;
+
+    /** Membrane capacitance in pF. */
+    double c_m_;
+
+    /** Refractory period in ms. */
+    double t_ref_;
+
+    /** Resting potential in mV. */
+    double E_L_;
+
+    /** External DC current [pA] */
+    double I_e_;
+
+    /** Threshold, RELATIVE TO RESTING POTENTAIL(!).
+        I.e. the real threshold is U_th_ + E_L_. */
+    double U_th_;
+
+    /** Lower bound, RELATIVE TO RESTING POTENTAIL(!).
+        I.e. the real lower bound is U_min_+E_L_. */
+    double U_min_;
+
+    /** Reset potential.
+        At threshold crossing, the membrane potential is reset to this value.
+        Relative to resting potential. */
+    double U_reset_;
+
+    Parameters_(); //!< Sets default parameter values
+
+    void get( DictionaryDatum& ) const; //!< Store current values in dictionary
+    double set( const DictionaryDatum& ); //!< Set values from dicitonary
+  };
+
+  // ----------------------------------------------------------------
+
+  /**
+   * State variables of the model.
+   */
+  struct State_
+  {
+    double y0_;       //!< External input current
+    double I_syn_ex_; //!< Exc. exponential current
+    double I_syn_in_; //!< Inh. exponential current
```

Comment:
Maybe add a comment about which values correspond to y1.

Reply:
I will leave it as it is (but fixed corresponding comments in `Variables_`,  since `y1_` is not used in this model.